src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -355,6 +351,23 @@ public AgentSettings LoadSettings()
             var serviceControlManager = HostContext.GetService<ILinuxServiceControlManager>();
```

Comment:
earlier this code was not getting executed if runAsService is false But this gets executed now.